common/achievement/get.ts

```diff
@@ -147,44 +147,49 @@ const assembleAchievementData = async (userAchievements: achievements_with_templ
     let currentAchievementIndex = userAchievements.findIndex((ua) => !ua.achievedAt);
     currentAchievementIndex = currentAchievementIndex >= 0 ? currentAchievementIndex : userAchievements.length - 1;
 
-    const achievementContext = transformPrismaJson(
-        user,
-        userAchievements[currentAchievementIndex].relation,
-        userAchievements[currentAchievementIndex].context as Prisma.JsonObject
-    );
-    const currentAchievementTemplate = renderAchievementWithContext(userAchievements[currentAchievementIndex], achievementContext);
-
     const achievementTemplates = await prisma.achievement_template.findMany({
-        where: { group: currentAchievementTemplate.group, isActive: true },
+        where: { group: userAchievements[currentAchievementIndex].template.group, isActive: true },
         orderBy: { groupOrder: 'asc' },
     });
 
     const state: achievement_state = getAchievementState(userAchievements, currentAchievementIndex);
 
     const isNewAchievement = state === achievement_state.COMPLETED && !userAchievements[currentAchievementIndex].isSeen;
 
+    const condition = userAchievements[currentAchievementIndex].recordValue
+        ? userAchievements[currentAchievementIndex].template.condition.replace(
+              'recordValue',
+              (userAchievements[currentAchievementIndex].recordValue + 1).toString()
+          )
+        : userAchievements[currentAchievementIndex].template.condition;
```

Comment:
because we can have variables in our conditions, so we need to replace those before using the condition inside the evaluateAchievement Function.

We use the evaluate achievement function inside the get request, because we need to reevaluate achievements, that have not been updated, due to not triggering events (not participating in meetings -> no meeting streak).

Reply:
I'm not sure if this makes sense. Here, we are replacing the `recordValue` text with a number and not any variable.
The same happens within the evaluate function already.

Reply:
This is still open.

Reply:
Now I got it! This is more or less unnecessary, since evaluate from swan.parse already replaces the record value. Thank you!